examples/script/concolic.py

```diff
@@ -272,31 +249,15 @@ def main():
         log('get constraint set from queue, queue size: {}'.format(q.qsize()))
         cons = q.get()
         inp = input_from_cons(cons, datas)
-        log('generated input from constraints: {}'.format(inp2ints(inp)))
-
-        log('running generate input concretely')
-        trc = concrete_run_get_trace(inp)
-        paths +=1 
-
-        log('doing symex on new generated input')
-        newcons, datas = symbolic_run_get_cons(trc)
-
-        # hmmm ideally do some smart stuff so we don't have to check if the
-        # constraints are unsat. something like the compare the constraints set
-        # which you used to generate the input, and the constraint set you got
-        # from the symex. sounds pretty hard
-        #
-        # but maybe a dumb way where we blindly permute the constraints
-        # and just check if they're sat before queueing will work
-
-        to_queue = get_new_constrs_for_queue(cons, newcons)
-
-        log('permuting constraints and adding {} constraints sets to queue'.format(len(to_queue)))
+        to_queue, new_datas = concrete_input_to_constraints(inp, cons)
+        if len(new_datas) > 0:
```

Comment:
oh cool, is this change meant to handle the case when a discovered path gets more symbolic input? if so, how was it tested?

Reply:
This was more of a refactoring from previous code to wrap `symbolic_run_get_cons`/`get_new_constrs_for_queue` in a function that was made sure to capture side effects. Testing, just made sure to run it on the binaries I was using with the original script with and made sure it found the same states and produced identical output.